File: frontend/src/features/dashboard/applications/models/JobApplicationDto.ts
import type { Position } from "./Position";
import type { Status } from "./Status";


export interface JobApplicationDto {
    id: string;
    name: string;
    city: string;
    status: Status;
    position: Position;
    isFavorite: boolean;
    appliedAt: string;
    note: string;
    isStale: boolean;
}

File: appliboard/src/main/java/com/backend/appliboard/features/job_application/dto/JobApplicationDto.java
package com.backend.appliboard.features.job_application.dto;

import com.backend.appliboard.features.job_application.Position;
import com.backend.appliboard.features.job_application.Status;

import java.time.Instant;
import java.util.UUID;

public record JobApplicationDto(
        UUID id,
        String name,
        String city,
        Status status,
        Position position,
        Boolean isFavorite,
        Instant appliedAt
) {
}
